tst/EnergyPlus/unit/UnitVentilator.unit.cc

```diff
@@ -272,4 +273,81 @@ TEST_F(EnergyPlusFixture, UnitVentilatorCalcMdotCCoilCycFanTest)
+    int constexpr variablePercent = 1;
+    int constexpr fixedTemperature = 2;
+    int constexpr fixedOAControl = 3;
+
+    // test fixed OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = fixedOAControl;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow,
+                state->dataSize->FinalZoneSizing(state->dataSize->CurZoneEqNum).MinOA,
+                0.001);
+
+    // test variable percent OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = variablePercent;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(
+        state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow, state->dataUnitVentilators->UnitVent(UnitVentNum).MaxAirVolFlow, 0.001);
+
+    // test fixed temperature OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = fixedTemperature;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(
+        state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow, state->dataUnitVentilators->UnitVent(UnitVentNum).MaxAirVolFlow, 0.001);
+
+    // retest without HVAC sizing object
+    state->dataUnitVentilators->UnitVent(UnitVentNum).HVACSizingIndex = 0;
+
+    // test fixed OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = fixedOAControl;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow,
+                state->dataSize->FinalZoneSizing(state->dataSize->CurZoneEqNum).MinOA,
+                0.001);
+
+    // test variable percent OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = variablePercent;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(
+        state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow, state->dataUnitVentilators->UnitVent(UnitVentNum).MaxAirVolFlow, 0.001);
+
+    // test fixed temperature OA control
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow = DataSizing::AutoSize;
+    state->dataUnitVentilators->UnitVent(UnitVentNum).OAControlType = fixedTemperature;
+    UnitVentilator::SizeUnitVentilator(*state, UnitVentNum);
+
+    EXPECT_NEAR(
+        state->dataUnitVentilators->UnitVent(UnitVentNum).OutAirVolFlow, state->dataUnitVentilators->UnitVent(UnitVentNum).MaxAirVolFlow, 0.001);
+}
```

Comment:
This is a great unit test.  Testing several different permutations.  Good stuff!